sklearn_numba_dpex/kmeans/kernels/compute_labels.py

```diff
@@ -36,12 +36,16 @@ def make_label_assignment_fixed_window_kernel(
     if work_group_size != input_work_group_size:
         work_group_size = centroids_window_height * sub_group_size
 
-    elif centroids_window_height * sub_group_size != work_group_size:
+    if (work_group_size == input_work_group_size) and (
+        (centroids_window_height * sub_group_size) != work_group_size
+    ):
         raise ValueError(
             "Expected work_group_size to be a multiple of sub_group_size but got "
             f"sub_group_size={sub_group_size} and work_group_size={work_group_size}"
         )
 
+    work_group_shape = (centroids_window_height, window_n_centroids)
```

Comment:
```suggestion
    work_group_shape = (centroids_window_height, centroids_window_width)
```

Reply:
or let's just drop `centroids_window_width` variable ? (that was my choice for other files, I probably forgot this one)